controllers/daemonset.go

```diff
@@ -171,44 +154,91 @@ func (dc *daemonSetGenerator) SetAsDesired(ds *appsv1.DaemonSet, image string, m
 		},
 	}
 
-	if mod.Spec.DevicePlugin != nil {
-		devicePlugin := *mod.Spec.DevicePlugin
-		devicePlugin.Name = "device-plugin"
+	return dc.constructDaemonSet(ctx, ds, &mod, mod.Spec.DriverContainer, "driver-container", image, standardLabels,
+		nodeSelector, driverContainerVolumeMounts, volumes, false)
+}
 
-		if devicePlugin.SecurityContext == nil {
-			devicePlugin.SecurityContext = &v1.SecurityContext{}
-		}
+func (dc *daemonSetGenerator) SetDevicePluginAsDesired(ctx context.Context, ds *appsv1.DaemonSet, mod *ootov1alpha1.Module) error {
+	if ds == nil {
+		return errors.New("ds cannot be nil")
+	}
+
+	if mod.Spec.DevicePlugin == nil {
+		return errors.New("device plugin in module should not be nil")
+	}
 
-		devicePlugin.SecurityContext.Privileged = &varTrue
+	standardLabels := map[string]string{
+		constants.ModuleNameLabel: mod.Name,
+	}
 
-		devicePluginsVolumeMount := v1.VolumeMount{
+	containerVolumeMounts := []v1.VolumeMount{
+		{
 			Name:      kubeletDevicePluginsVolumeName,
 			MountPath: kubeletDevicePluginsPath,
-		}
-
-		devicePlugin.VolumeMounts = append(devicePlugin.VolumeMounts, devicePluginsVolumeMount)
-
-		containers = append(containers, devicePlugin)
+		},
+	}
 
-		devicePluginsVolume := v1.Volume{
+	hostPathDirectory := v1.HostPathDirectory
+	dsVolumes := []v1.Volume{
+		{
 			Name: kubeletDevicePluginsVolumeName,
 			VolumeSource: v1.VolumeSource{
 				HostPath: &v1.HostPathVolumeSource{
 					Path: kubeletDevicePluginsPath,
 					Type: &hostPathDirectory,
 				},
 			},
-		}
+		},
+	}
+
+	return dc.constructDaemonSet(ctx, ds, mod, *mod.Spec.DevicePlugin, "device-plugin", "", standardLabels,
+		mod.Spec.Selector, containerVolumeMounts, dsVolumes, true)
+}
+
+func (dc *daemonSetGenerator) constructDaemonSet(ctx context.Context,
+	ds *appsv1.DaemonSet,
+	mod *ootov1alpha1.Module,
+	container v1.Container,
```

Comment:
Do we really need it? It is already part of the module object passed to that function.

Reply:
this is a common function, used both for creating DevicePlugin daemonset and DrvierContainer daemonset. If we don't pass the container, then we need to pass a flag, that will tell the code which container to use.  So i prefer to pass the container directly, instead of passing the flag and adding if clause into the code